pkg/instrumentation/sdk.go

```diff
@@ -263,8 +263,22 @@ func (i *sdkInjector) injectCommonSDKConfig(ctx context.Context, otelinst v1alph
 		}
 	}
 
+	// get existing resources env var and parse it into a map
+	existingRes := map[string]string{}
+	existingResourceEnvIdx := getIndexOfEnv(pod.Spec.Containers[appIndex].Env, constants.EnvOTELResourceAttrs)
+	if existingResourceEnvIdx > -1 {
+		existingResArr := strings.Split(pod.Spec.Containers[appIndex].Env[existingResourceEnvIdx].Value, ",")
+		for _, kv := range existingResArr {
+			keyValueArr := strings.Split(strings.TrimSpace(kv), "=")
+			if len(keyValueArr) != 2 {
+				continue
+			}
+			existingRes[keyValueArr[0]] = keyValueArr[1]
+		}
+	}
+
 	// Some attributes might be empty, we should get them via k8s downward API
-	if resourceMap[string(semconv.K8SPodNameKey)] == "" {
+	if existingRes[string(semconv.K8SPodNameKey)] == "" && resourceMap[string(semconv.K8SPodNameKey)] == "" {
```

Comment:
Not sure I understand...why isn't `resourceMap` sufficient to determine if the k8s pod name key is missing?

Reply:
for k, v := range otelinst.Spec.Resource.Attributes {
	if !existingRes[k] {
		res[k] = v
	}
}
If the attribute doesn't exist in existingRes, then the res will be updated. So when the second injection happens, res doesn't have the semconv.K8SPodNameKey key in it as it was existing value. So just checking the resourceMap[string(semconv.K8SPodNameKey)] == "" will be true and the key is added again